src/mon/OSDMonitor.cc

```diff
@@ -519,6 +519,10 @@ int OSDMonitor::reweight_by_utilization(int oload,
 	if (*q >= (int)pgs_by_osd.size())
 	  pgs_by_osd.resize(*q);
 	if (pgs_by_osd[*q] == 0) {
+          if (osdmap.crush->get_item_weightf(*q) <= 0) {
```

Comment:
@tchaikov  Upgraded https://github.com/ceph/ceph/pull/8941/commits/6effa004dd060f3e1c0a5e057d642a667f057d28 to a safer way.